lm_eval/tasks/clindiagnoses/utils.py

```diff
@@ -0,0 +1,15 @@
+prompt_template="""A partir del caso clínico que se expone a continuación, tu tarea es la siguiente.
+Como médico experto, tu tarea es la de diagnosticar al paciente en base al caso clínico. Responde únicamente con el diagnóstico para el paciente de forma concisa.
+Caso clínico: {caso_clinico}
+"""
+
+system_prompt = "Eres un experto en medicina que realiza diagnósticos en base a casos clínicos."
+
+
+def case_to_rawtext(sample: dict):
```

Comment:
Me parece un tanto complejo el prompt en comparación con los que he visto, igual quitaría el "System:" "User:" "Assistant" y dejaría solo el `prompt_template` no? O sí que has encontrado ejemplos así? Lo digo de cara a que te acepten la PR en el repo principal que es el objetivo, tenemos el fork solo para que no nos bloqueen las reviews de los maintainers del harness

Reply:
De hecho ya han implementado la opción del chat template o sea que definitivamente dejaría solo lo que llamas "prompt_template"

Reply:
¿Cómo es para aplicar el chat template? Lo ideal sería usar eso, directamente sin hacer esto que al final es definir un prompt custom cuando cada tokenizador-modelo tiene el suyo.

Reply:
Han implementado un parámetro (bool) que se le pasa a la función de evaluación, es independiente de la tarea, lo añadiremos a la interfaz antes de evaluar los modelos IT, te parece bien entonces dejar solo el prompt con la pregunta?

Reply:
Habría que crear entonces una lista de mensajes estilo:

```python
messages=[{"role": "system", "content": "Eres un experto..."}, {"role": "input", "content": el artículo}, {"role": "user", "content": La pregunta}]
```
Y a esa lista de mensajes se le aplica el chat template:

```python
tokenizer.apply_chat_template(messages, tokenize=True, add_generation_prompt=True)
```

De esta forma se adaptaría al chat template de cada modelo

Reply:
Creo que a nivel tarea no se puede configurar, es solo a nivel modelo, no he encontrado ninguna tarea especificando los role. Igualmente de momento la leaderboard no lo admite o sea que si te parece bien evaluamos de momento los modelos generativos con el prompt_template y cuando la semana que viene implementemos el chat template para evaluar IT revisamos esto? :)